tests/v2/validation/snapshot/snapshot.go

```diff
@@ -4,148 +4,186 @@ import (
 	require.NoError(t, err)
 
-	clusterObject, _, err := clusters.GetProvisioningClusterByName(adminClient, clusterName, namespace)
-	require.NoError(t, err)
-
-	initialKubernetesVersion := clusterObject.Spec.KubernetesVersion
 	steveID := fmt.Sprintf("%s/%s", clusterObject.Namespace, clusterObject.Name)
 	err = clusters.WatchAndWaitForCluster(client, steveID)
 	require.NoError(t, err)
 
 	podResults, podErrors := pods.StatusPods(client, clusterID)
 	assert.NotEmpty(t, podResults)
 	assert.Empty(t, podErrors)
-	require.NoError(t, err)
+
+	initialKubernetesVersion := clusterObject.Spec.KubernetesVersion
 	require.Equal(t, initialKubernetesVersion, clusterObject.Spec.KubernetesVersion)
 
 	snapshotToRestore, err := provisioning.VerifySnapshots(client, localClusterID, clusterName, etcdNodeCount+len(existingSnapshots))
 	require.NoError(t, err)
 
-	deploymentResponse, err = createDeployment(steveclient, WorkloadNamePostBackup)
+	postBackupDeployment := workloads.NewDeploymentTemplate(WorkloadNamePostBackup, defaultNamespace, podTemplate, isCattleLabeled, nil)
+	postBackupService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAppendName + WorkloadNamePostBackup,
+			Namespace: defaultNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name: port,
+					Port: 80,
+				},
+			},
+			Selector: deployment.Spec.Template.Labels,
+		},
+	}
+
+	postDeploymentResp, _, err := workloads.CreateDeployment(steveclient, WorkloadNamePostBackup, postBackupDeployment, postBackupService)
 	require.NoError(t, err)
 
-	err = workloads.VerifyDeployment(steveclient, deploymentResponse)
+	err = workloads.VerifyDeployment(steveclient, postDeploymentResp)
 	require.NoError(t, err)
-	require.Equal(t, WorkloadNamePostBackup, deploymentResponse.ObjectMeta.Name)
-	var restoreRKEConfig string
+	require.Equal(t, WorkloadNamePostBackup, postDeploymentResp.ObjectMeta.Name)
+
 	if upgrade {
-		clusterObject, clusterResponse, err := clusters.GetProvisioningClusterByName(adminClient, clusterName, namespace)
+		clusterObject, clusterResponse, err := clusters.GetProvisioningClusterByName(client, clusterName, namespace)
 		require.NoError(t, err)
 
-		var upgradeKubernetesVersion string
-		if strings.Contains(initialKubernetesVersion, "rke2") {
-			defaultVersion, err := kubernetesversions.Default(adminClient, clusters.RKE2ClusterType.String(), nil)
+		if strings.Contains(initialKubernetesVersion, RKE2) {
+			defaultVersion, err := kubernetesversions.Default(client, clusters.RKE2ClusterType.String(), nil)
 			upgradeKubernetesVersion = defaultVersion[0]
 			require.NoError(t, err)
 		} else {
-			defaultVersion, err := kubernetesversions.Default(adminClient, clusters.K3SClusterType.String(), nil)
+			defaultVersion, err := kubernetesversions.Default(client, clusters.K3SClusterType.String(), nil)
 			upgradeKubernetesVersion = defaultVersion[0]
 			require.NoError(t, err)
 		}
```

Comment:
It doesn't need to be part of these changes, but I wanted to plant the seed in your mind if it makes sense to be able to specify an upgradeVersion in the config itself instead of just taking the default -- enabling us to do additional upgrade scenarios like v1.25.13+rke2r1 --> v1.25.14+rke2r1.

Reply:
That's a fair point. I have in place now an `upgradeKubernetesVersion` that is taken as a parameter to the function overall. I will update the check so that if that is left blank, then get the default version. Otherwise, upgrade to that specified version.

Reply:
Resolved